tests/syntax/test_parser.py

```diff
@@ -336,3 +336,849 @@ def test_specifier_apparently_facing_from(self):
+                RelativePositionOp(
+                    RelativePositionOp(Name("A", Load()), Name("B", Load())),
+                    Name("C", Load()),
+                ),
+            ),
+            (
+                "relative position of A from relative position of B from C",
+                RelativePositionOp(
+                    Name("A", Load()),
+                    RelativePositionOp(Name("B", Load()), Name("C", Load())),
+                ),
+            ),
+            (
+                "relative position of A << B from C",
+                RelativePositionOp(
+                    BinOp(Name("A", Load()), LShift(), Name("B", Load())),
+                    Name("C", Load()),
+                ),
+            ),
+            (
+                "relative position of A from B << C",
+                BinOp(
+                    RelativePositionOp(Name("A", Load()), Name("B", Load())),
+                    LShift(),
+                    Name("C", Load()),
+                ),
+            ),
+            (
+                "relative position of A + B from C",
+                RelativePositionOp(
+                    BinOp(Name("A", Load()), Add(), Name("B", Load())),
+                    Name("C", Load()),
+                ),
+            ),
+            (
+                "relative position of A from B + C",
+                RelativePositionOp(
+                    Name("A", Load()),
+                    BinOp(Name("B", Load()), Add(), Name("C", Load())),
+                ),
+            ),
+            (
+                "relative position of A << B",
+                BinOp(
+                    RelativePositionOp(Name("A", Load())), LShift(), Name("B", Load())
+                ),
+            ),
+            (
+                "relative position of A + B",
+                RelativePositionOp(BinOp(Name("A", Load()), Add(), Name("B", Load()))),
+            ),
+        ],
+    )
+    def test_relative_position_precedence(self, code, expected):
+        mod = parse_string_helper(code)
+        stmt = mod.body[0].value
+        assert dump(stmt, annotate_fields=False) == dump(
+            expected, annotate_fields=False
+        )
+
+    def test_relative_heading(self):
+        mod = parse_string_helper("relative heading of x")
+        stmt = mod.body[0]
+        match stmt:
+            case Expr(RelativeHeadingOp(Name("x"))):
+                assert True
+            case _:
+                assert False
+
+    def test_relative_heading_from(self):
+        mod = parse_string_helper("relative heading of x from y")
+        stmt = mod.body[0]
+        match stmt:
+            case Expr(RelativeHeadingOp(Name("x"), Name("y"))):
+                assert True
+            case _:
+                assert False
+
+    def test_relative_heading_precedence(self):
+        # This is incorrect code, but asserts the operator has the right precedence
+        mod = parse_string_helper("relative heading of x or y")
```

Comment:
We ultimately want this to parse as `(relative heading of x) or y`, though; are you planning to do that in a later PR, preserving the current (bad) behavior of the `TokenTranslator` for now?

Reply:
Are you sure `(relative heading of x) or y` is the desired result of this program? The current setup is that `relative heading of` has precedence higher than bitwise shifts and lower than addition/subtraction. 

https://docs.python.org/3/reference/expressions.html#operator-precedence

Because `or` has further lower precedence than shifts, currently it parses as `relative heading of (x or y)`. 

Maybe we should create a table and agree on the desired precedence of each of the operators. Changing the precedence itself should be relatively easy (just move rules in the grammar and adjust test cases)